### ISSUE TITLE
Initiator is not well defined

### ISSUE THREAD

**Issue by yoavweiss** [w3c/resource-timing#69]
I've been looking at specific ways to add more detailed initiator info that would enable getting the dependency tree info from resource-timing (as briefly discussed during TPAC).

When digging deeper into the spec and the current definitions, I found some unfortunate surprises:
- The term "initiator" is not really defined in the spec
- `initiatorType` sometime points at the type of the resource itself, and sometimes points at the type of the initiator (specifically, with the "css" initiator type)
- Fetch contains the concept of "[initiator](https://fetch.spec.whatwg.org/#concept-request-initiator)", which is not related to what we need here in any way.

Furthermore, for the purposes of a dependency tree, initiator definition needs to vary from the one practiced with `initiatorType` (and match more closely what happens in the various browser dev tools):
- If an element triggered a resource download and was part of the HTML's markup, the "dependency initiator" should be the HTML itself
- If an element triggered a resource download, but was added to the HTML by a JS file, the "dependency initiator" should be the JS resource (or maybe JS resources, if the addition was the result of multiple JS resources executing)
- Cases which now get "css" as initiator type, the "dependency initiator" should point to the CSS resource.
- If a download was triggered by XHR/Fetch, the "dependency initiator" should point to the JS resource(s).

I think that for the purposes of `initiatorType`, we'd better define the behavior as it is currently shipped.

For the purposes of the "dependency initiator", we probably need to define something slightly different. As `initiatorType` is already shipped and the term "initiator" seems to be overloaded, we probably also want to find a new name.

**Comment by igrigorik** [comment]
> I think that for the purposes of initiatorType, we'd better define the behavior as it is currently shipped

Agreed, I don't think we can change it significantly. That said, what would you propose we change or update with respect to current definition?

Aside: I propose we focus this thread on the above and resolve it separately from a design discussion on what we could/should do to tackle the "dependency tree" use case. For the latter, let's open a new thread and describe the use case(s) and what's needed to enable them.

**Comment by igrigorik** [comment]
Taking a second pass at this... 

> I think that for the purposes of initiatorType, we'd better define the behavior as it is currently shipped.

Spec: https://www.w3.org/TR/resource-timing-2/#dom-performanceresourcetiming-initiatortype

> On getting, the initiatorType attribute must return one of the following DOMString:
> 
> - The same value as the localName of that element [DOM], if the initiator is an element.
> - "css", if the initiator is a CSS resource downloaded by the url() syntax [CSS-SYNTAX-3], such as @import url() or background: url().
> - "xmlhttprequest", if the initiator is an XMLHttpRequest object [XMLHttpRequest].

One notable omission in above definition is what to return if none of the above match. For example, a quick test with fetch() shows that Chrome returns an empty string, while Firefox returns "other" - @toddreifsteck can you try it on Edge?

I think the minimum the need to do here is agree on a fallback. It may also not be a bad idea to add a clause for Fetch.. report "fetch"? Outside of that, I propose we keep the definition as is for L2 as we already have three shipping implementations, and revisit it in L3 - see https://github.com/w3c/resource-timing/issues/8.

**Comment by igrigorik** [comment]
@toddreifsteck @nolanlawson trying to test this on Edge.. but I'm not seeing any fetch-initiated requests in the perf timeline at all. Is this a bug, or am I doing something silly?

**Comment by toddreifsteck** [comment]
@igrigorik Bug in Edge 14. Fix is understood and will be shipped to Insider builds of Edge 15 in ~2-4 weeks.

**Comment by igrigorik** [comment]
@toddreifsteck gotcha, thanks. Do you know what you guys set the initiatorType value to for fetch-initiated requests?

**Comment by toddreifsteck** [comment]
We plan to use 'fetch' for initatorType for fetch and mostly like 'beacon' or 'sendBeacon' (depending on other browser's implementation) for sendBeacon.

**Comment by igrigorik** [comment]
Resolved via #80, closing.

**Comment by MasonM** [comment]
I stumbled on this issue while trying to figure out what `initiatorType` means, since the current version version of the spec doesn't define what it means or what the possible values are. The changes in https://github.com/w3c/resource-timing/pull/80 have what was I was looking for, but that was reverted in https://github.com/w3c/resource-timing/pull/261. Was that information moved somewhere else? The latter PR talks about the [fetch spec](https://fetch.spec.whatwg.org/), but that spec doesn't define `initiatorType` either.
